pkg/security/ptracer/proc.go

```diff
@@ -0,0 +1,286 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package ptracer
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/security/proto/ebpfless"
+	"github.com/DataDog/gopsutil/process"
+)
+
+func isKThread(ppid, pid int32) bool {
+	return ppid == 2 || pid == 2
+}
+
+func listNotTracedProcess(tracerPID int32) (map[int32]*process.Process, error) {
+	pids, err := process.Pids()
+	if err != nil {
+		return nil, err
+	}
+
+	processes := make(map[int32]*process.Process)
+
+	for _, pid := range pids {
+		proc, err := process.NewProcess(pid)
+		if err != nil {
+			continue
+		}
+		processes[pid] = proc
+	}
+
+	var traced []int32
+
+	isTraced := func(proc *process.Process) bool {
+		for {
+			if proc.Pid == tracerPID || slices.Contains(traced, proc.Pid) {
+				return true
+			}
+
+			ppid, err := proc.Ppid()
+			if err != nil {
+				// if we can get the ppid ignore the process as it can be a race
+				return true
+			}
+
+			if ppid == 1 || ppid == 0 {
+				return false
+			}
```

Comment:
Processes with ppid 1/0 should already be traced through the entry point right ? I'm not sure I understand this check